auslib/db.py

```diff
@@ -1617,8 +1657,10 @@ def assertOptionsExist(self, permission, options):
                 raise ValueError('Unknown option "%s" for permission "%s"' % (opt, permission))
 
     def getAllUsers(self, transaction=None):
-        res = self.select(columns=[self.username], distinct=True, transaction=transaction)
-        return [r['username'] for r in res]
+        users = self.select(columns=[self.username], distinct=True, transaction=transaction)
+        users = set([u["username"] for u in users])
+        users.update(self.user_roles.getAllUsers(transaction))
```

Comment:
Good call. There's no use case I can think of for having a Role without a Permission. Such a User wouldn't be able to do any harm (Signoffs are likely to require a "signoff" permission), but it's a weird state to be in. I'll look at this, and at cascading the deletion of a User's last permission.